rmpv/src/ext/se.rs

```diff
@@ -787,20 +819,26 @@ impl ser::SerializeMap for DefaultSerializeMap {
     }
 }
 
-impl SerializeStruct for SerializeVec {
+impl SerializeStruct for DefaultSerializeStruct {
     type Ok = Value;
     type Error = Error;
 
     #[inline]
-    fn serialize_field<T: ?Sized>(&mut self, _key: &'static str, value: &T) -> Result<(), Error>
+    fn serialize_field<T: ?Sized>(&mut self, key: &'static str, value: &T) -> Result<(), Error>
         where T: Serialize
     {
-        ser::SerializeSeq::serialize_element(self, value)
+        self.map.push((to_value(key)?, to_value(value)?));
```

Comment:
You might want to call `to_value_named(value)` for value here if `named` is set to `true` to ensure any struct values get serialised with key names.

Similar kind of refactoring needs to be done for `SerializeVec`, `DefaultSerializeMap` etc. to ensure any struct values are encoded with key names if `Serializer` has `named_structs` set to `true`